packages/manager/apps/pci-private-network/src/data/services/services.ts

```diff
@@ -0,0 +1,47 @@
+import { TNetworkCreationData } from '@/types/network.type';
+import { NewPrivateNetworkForm } from '@/types/private-network-form.type';
+import { createPrivateNetwork } from '@/data/api/networks';
+import { assignGateway, enableSnatOnGateway } from '@/data/api/gateway';
+import { useCheckPrivateNetworkCreationStatus } from '../hooks/networks/useNetworks';
+
+export const handleCreatePrivateNetwork = async (
```

Comment:
the conception of this method seems weird, do you think you rewrite it in functional programming ?